### PR TITLE
Modified the property TimeInterval of GenerateEventsFilter.

### PR DESCRIPTION
**Pull request by wdzhou**
Such that it can take a list of doubles instead of 1.
-   modified:   ../Mantid/Framework/Algorithms/src/GenerateEventsFilter.cpp
-   modified:   ../Mantid/Framework/Algorithms/test/GenerateEventsFilterTest.h
-   modified:   ../Mantid/Framework/Kernel/src/InternetHelper.cpp:
  the cast of osstringstream to osstream on mac osx.  Modify it
  only to make the compiling pass.

### CODE REVIEW

**Comment by wdzhou** [comment]
Fixes issue [#11112](http://mantidproject.org/mantid/ticket/11112)

The changes is to enable GenerateEventsFilter to generate splitters with various time intervals. Besides the documentation of this algorithm and unit tests are updated for the new feature. 

For tester

1.   Check whether any system test and unit test gets failed due to the change of input property 'TimeInterval'; 

2.   Find an event Nexus file, such as PG3_9829_event.nxs in system test. Then run GenerateEventFilters by setting TimeInterval to a few numbers for various lengths of splitters. Then check whether the output SplitterWorkspace has the correct output. You may also run FilterEvents from the result with option 'SplitLogs' on. It is easy to check the sample logs such that they are split as you expect.

**Comment by martyngigg** [comment]
Could you update the pull request title remove the ticket number, following the guidelines [here](http://www.mantidproject.org/Git_Workflow).

File: Code/Mantid/docs/source/algorithms/GenerateEventsFilter-v1.rst [hunk 1]
```diff
@@ -81,9 +87,35 @@ this algorithm:
...
+respectively. 
+
+*TimeInterval* accepts an array of doubles.  
+If the array size is zero, then there will be one and only splitter will be 
+created from *StartTime* and *StopTime*.  
+If the size of the array is one, then all event splitters will have the same duration
+of time. 
+In general if the array is composed as :math:`t_1, t_2, \cdots, t_n`, 
+and :math:`T_0` is the run start time, 
+then the event splitters will have the time boudaries as
```
Comment (FedeMPouzols) [inline Code/Mantid/docs/source/algorithms/GenerateEventsFilter-v1.rst]: There's a small typo here: boudaries -> boundaries.

I think this is the type of minor thingy that can be fixed without requiring testing.

**Comment by FedeMPouzols** [comment]
Not really part of this ticket, but very related, I was checking the wiki page on event filtering (http://www.mantidproject.org/EventFiltering) and I noticed that there are many outdated links. It would be a good idea if someone familiar with this would have a look at it and bring it up to date.

**Comment by FedeMPouzols** [comment]
It all looks good to me, and well documented. GenerateEventFilters and FilterEvents behave well. Unit and system tests pass.

**Pull request merged**